### PR TITLE
Use sidenav instead of md-menu for collapsed navbar

### PR DESCRIPTION
**Pull request by hoangviet1993**
I decided to __not__ make the Oppia foundation logo in the sidenav a button. Might reverse my decision later though if people demands it to be a button.

GIF:
iPhone 

Pixel2

### CODE REVIEW

**Comment by hoangviet1993** [comment]
Changes in this PR are live at:
https://mail-app-214415.appspot.com/#!/

**Comment by hoangviet1993** [comment]
Discovered a bug with the sidenav not covering the entire screen on mobile.

Currently investigating and will try to patch within this PR.

File: static/index.html [hunk 1]
```diff
@@ -42,13 +42,13 @@
...
-                       md-whiteframe-4dp
-                       oppia-foundation-green-background"
-                md-component-id="right">
-      <side-bar></side-bar>
-    </md-sidenav>
     <div class="outer">
+      <md-sidenav class="md-sidenav-right
+                                           md-whiteframe-4dp
+                                           oppia-foundation-green-background"
+        md-component-id="right">
```
Comment (seanlip) [inline static/index.html]: Ditto.

File: static/index.html [hunk 2]
```diff
@@ -42,6 +43,12 @@
 
   <body layout="row" aria-label="Oppia Foundation Page">
     <div class="outer">
+      <md-sidenav class="md-sidenav-right
+                                           md-whiteframe-4dp
```
Comment (seanlip) [inline static/index.html]: Indentation seems off. Make sure to align properly with previous row.

File: static/pages/navbar/NavigationBarDirective.js [hunk 3]
```diff
@@ -17,7 +17,10 @@ oppiaFoundationWebsite.directive('navigationBar', [function() {
     restrict: 'E',
     scope: {},
     templateUrl: '/pages/navbar/navbar.html',
-    controller: ['$scope', function($scope) {
+    controller: ['$scope', '$mdSidenav', function($scope, $mdSidenav) {
+      $scope.open = function() {
```
Comment (seanlip) [inline static/pages/navbar/NavigationBarDirective.js]: Probably needs a better name: openSidebar?

File: static/pages/navbar/SideBarDirective.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppiaFoundationWebsite.directive('sideBar', [function() {
+  return {
+    restrict: 'E',
+    scope: {},
+    templateUrl: '/pages/navbar/sidebar.html',
+    controller: ['$scope', '$mdSidenav', function($scope, $mdSidenav) {
+      $scope.close = function() {
```
Comment (seanlip) [inline static/pages/navbar/SideBarDirective.js]: Ditto.

File: static/pages/navbar/sidebar.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<md-toolbar layout="row" style="padding-top: 10px;">
+    <img src="/assets/images/navbar/oppia_foundation_logo.png"
```
Comment (seanlip) [inline static/pages/navbar/sidebar.html]: indent by 2 here and below, not 4

**Comment by kevinlee12** [comment]
Hi @hoangviet1993, I tried it on my iOS device (it's a 5s) and the nav "X" disappears. However, it comes back on landscape mode.

**Comment by kevinlee12** [comment]
However, the behavior of the navbar looks fine.

**Comment by hoangviet1993** [comment]
Thanks @kevinlee12 , the bug is reproducible using Chrome's Dev Tool (ip5S).
Will be patched in a sec.

**Comment by kevinlee12** [comment]
It works on 5s now, thanks @hoangviet1993!

**Comment by rachelwchen** [comment]
Looks great! Thank you!

**Pull request merged**